e2e/aks_model.go

```diff
@@ -428,3 +458,57 @@ func updateSubnet(ctx context.Context, cluster *armcontainerservice.ManagedClust
 	}
 	return nil
 }
+
+func createAzureResourceWithBicep(ctx context.Context, t *testing.T, params map[string]interface{}, resourceGroup, filename, deploymentName string) error {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("failed to get current working directory: %v", err)
+	}
+
+	bicepFileName := fmt.Sprintf("%s.bicep", filename)
+	bicepFilePath := filepath.Join(currentDir, "azure-resources", bicepFileName)
+	cmd := exec.Command("az", "bicep", "build", "--file", bicepFilePath)
+	if err := cmd.Run(); err != nil {
+		log.Fatalf("failed to run az bicep build: %v", err)
+	}
+
+	jsonFileName := fmt.Sprintf("%s.json", filename)
+	jsonFilePath := filepath.Join(currentDir, "azure-resources", jsonFileName)
+	if _, err := os.Stat(jsonFilePath); os.IsNotExist(err) {
+		log.Fatalf("JSON file was not created: %v", err)
+	}
+	jsonFile, err := os.ReadFile(jsonFilePath)
```

Comment:
if we end up just using a static ARM template then I'd recommend using a `go:embed` to have access to the template content rather than manually reading the file

Reply:
it doesn't matter for e2e, we don't compile a test binary and ship it.

Embedding is probably just going to add an extra layer.

Reply:
also, the file is probably not available yet at the moment you would embed it